comp/metadata/host/utils/tags_test.go

```diff
@@ -100,61 +86,52 @@ func TestMarshalEmptyHostTags(t *testing.T) {
 	}
 
 	marshaled, _ := json.Marshal(tags)
-
 	// `System` should be marshaled as an empty list
 	assert.Equal(t, string(marshaled), `{"system":[]}`)
 }
 
 func TestCombineExtraTags(t *testing.T) {
-	ctx := context.Background()
-	mockConfig := config.Mock(t)
-	mockConfig.Set("autoconfig_from_environment", false)
-	defer mockConfig.Set("autoconfig_from_environment", true)
-
+	mockConfig, ctx := setupTest(t)
 	mockConfig.Set("tags", []string{"tag1:value1", "tag2", "tag4"})
 	mockConfig.Set("extra_tags", []string{"tag1:value2", "tag3", "tag4"})
-	defer mockConfig.Set("tags", nil)
-	defer mockConfig.Set("extra_tags", nil)
 
-	hostTags := GetHostTags(ctx, false)
+	hostTags := GetHostTags(ctx, false, mockConfig)
 	assert.NotNil(t, hostTags.System)
 	assert.Equal(t, []string{"tag1:value1", "tag1:value2", "tag2", "tag3", "tag4"}, hostTags.System)
 }
 
 func TestHostTagsCache(t *testing.T) {
-	ctx := context.Background()
-	mockConfig := config.Mock(t)
-	mockConfig.Set("autoconfig_from_environment", false)
-	defer mockConfig.Set("autoconfig_from_environment", true)
-
+	mockConfig, ctx := setupTest(t)
 	mockConfig.Set("collect_gce_tags", false)
 
 	fooTags := []string{"foo1:value1"}
 	var fooErr error
+	nbCall := 0
 
-	getProvidersDefinitionsFunc = func() map[string]*providerDef {
+	getProvidersDefinitionsFunc = func(config.ConfigReader) map[string]*providerDef {
 		return map[string]*providerDef{
 			"foo": {
-				retries: 1,
+				retries: 2,
```

Comment:
Just curious - why did we need to change this from `1` -> `2`?

Reply:
Setting it to 1 doesn't actually test the retry mechanism since we stop after 1 try. I wanted to make sure the test call the callback multiple time if configured to.